src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts

```diff
@@ -0,0 +1,184 @@
+      });
+      this.ZonegroupSub = this.rgwZonegroupService.list().subscribe((data: any) => {
+        this.rgwZonegroupCount = data['zonegroups'].length;
+      });
+      this.ZoneSUb = this.rgwZoneService.list().subscribe((data: any) => {
+        this.rgwZoneCount = data['zones'].length;
+      });
+      this.BucketSub = this.rgwBucketService.list().subscribe((data: any) => {
+        this.rgwBucketCount = data.length;
+      });
+      this.UserSub = this.rgwUserService.list().subscribe((data: any) => {
+        this.UserCount = data.length;
+      });
+      this.HealthSub = this.healthService.getClusterCapacity().subscribe((data: any) => {
+        this.objectCount = data['total_objects'];
+        this.totalPoolUsedBytes = data['total_pool_bytes_used'];
+      });
+    });
+    this.getPrometheusData(this.lastHourDateObject);
+  }
+
+  ngOnDestroy() {
+    this.interval.unsubscribe();
+    this.daemonSub.unsubscribe();
+    this.realmSub.unsubscribe();
+    this.ZonegroupSub.unsubscribe();
+    this.ZoneSUb.unsubscribe();
+    this.BucketSub.unsubscribe();
+    this.UserSub.unsubscribe();
+    this.HealthSub.unsubscribe();
+    if (this.timerGetPrometheusDataSub) {
+      this.timerGetPrometheusDataSub.unsubscribe();
+    }
+  }
+
+  getPrometheusData(selectedTime: any) {
+    this.prometheusService.ifPrometheusConfigured(() => {
+      if (this.timerGetPrometheusDataSub) {
+        this.timerGetPrometheusDataSub.unsubscribe();
+      }
+      this.timerGetPrometheusDataSub = timer(0, this.timerTime).subscribe(() => {
+        selectedTime = this.updateTimeStamp(selectedTime);
+
+        for (const queryName in queries) {
+          if (queries.hasOwnProperty(queryName)) {
+            const query = queries[queryName];
+            let interval = selectedTime.step;
+
+            if (query.includes('rate') && selectedTime.step < 20) {
+              interval = 20;
+            } else if (query.includes('rate')) {
+              interval = selectedTime.step * 2;
+            }
+
+            let intervalAdjustedQuery = query.replace(/\[(.*?)\]/g, `[${interval}s]`);
+
+            this.prometheusService
+              .getPrometheusData({
+                params: intervalAdjustedQuery,
+                start: selectedTime['start'],
+                end: selectedTime['end'],
+                step: selectedTime['step']
+              })
+              .subscribe((data: any) => {
+                if (data.result.length) {
+                  this.queriesResults[queryName] = data.result[0].values;
+                }
+                this.queriesResults[queryName].forEach((valueArray: string[]) => {
+                  if (valueArray.includes('NaN')) {
+                    const index = valueArray.indexOf('NaN');
+                    if (index !== -1) {
+                      valueArray[index] = '0';
+                  }
+                  }
+                })
+              });
+          }
```

Comment:
What's new/different in this method than in this other one:
https://github.com/ceph/ceph/blob/82ddbb6cd1fbf3c9f827fcc4c5470c8412b813c0/src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/dashboard/dashboard-v3.component.ts#L168-L203

BTW, we already have a Prometheus service with a `.getPrometheusData()`, but its implementation is naive:
https://github.com/ceph/ceph/blob/82ddbb6cd1fbf3c9f827fcc4c5470c8412b813c0/src/pybind/mgr/dashboard/frontend/src/app/shared/api/prometheus.service.ts#L27-L29

We should rather extract, encapsulate and reuse code, rather than duplicating and scattering it throughout the codebase.

Reply:
Sure, will try to re-use this one